flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/gsa/GatherSumApplyIteration.java

```diff
@@ -289,6 +300,11 @@ private GatherUdf(GatherFunction<VV, EV, M> gatherFunction, TypeInformation<Tupl
 
 		@Override
 		public void open(Configuration parameters) throws Exception {
+			try {
+				Collection<LongValue> numberOfVertices = getRuntimeContext().getBroadcastVariable("number of vertices");
+				this.gatherFunction.setNumberOfVertices(numberOfVertices.iterator().next().getValue());
+			} catch (Exception e) {
+			}
```

Comment:
@sewen thanks for the pointers.

What are your thoughts on adding a hasBroadcastVariable method to RuntimeContext? And elsewhere in that class?

The javadocs for getBroadcastVariable could note the exception as for getAccumulator.